### PR TITLE
#142 :sparkles: add retry ability into push for smoother monorepo rel…

### PR DESCRIPTION
**Pull request by d-ryan-ashcraft**
…eases

### CODE REVIEW

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PublishToPyPiRepoMojo.java [hunk 1]
```diff
@@ -90,6 +102,27 @@ public class PublishToPyPiRepoMojo extends AbstractHabushuMojo {
...
 
+    /**
+     * Specifies the number of times a push to the configured PyPI repository will be attempted before stopping (inclusive
+     * of the initial attempt). While this defaults to three and is fully configurable, it can be set to zero to never
+     * retry or set to any negative number for unlimited retries.  Unlimited retries will follow a fibonacci backoff
+     * interval.
+     */
+    @Parameter(property = "habushu.pyPiPushRetries", defaultValue = "3")
+    protected int pypiPushRetries = 3;
+
```
Comment (d-ryan-ashcraft) [inline habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PublishToPyPiRepoMojo.java:113]: I (Informational): purposely not putting these in the main readme.  It has gotten too complicated and is now less useful.  A ticket will flow behind this to simplify it overall and point to these types of supplementary options at the same time.  (TL;DR - we just copy the the comment into the README anyway, so the new process should just point to where additional options can be found while focusing on the big picture items in the README itself)

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PublishToPyPiRepoMojo.java [hunk 2]
```diff
@@ -90,6 +102,27 @@ public class PublishToPyPiRepoMojo extends AbstractHabushuMojo {
...
+    /**
+     * A multiplier to apply to the retry wait time (e.g., round of fibonacci).
+     */
+    @Parameter(property = "habushu.pypiPushRetryMultiplier", defaultValue = "15000")
+    protected long pypiPushRetryMultiplier;
+
+    /**
+     * Maximum time to wait for a retry, even if the backoff algorithm is above this threshold.
+     */
+    @Parameter(property = "habushu.pypiPushRetryMaxTimeout", defaultValue = "3")
```
Comment (epwilkins) [inline habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PublishToPyPiRepoMojo.java:123]: A: Might be worth specifying what unit is used for this field

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PublishToPyPiRepoMojo.java [hunk 3]
```diff
@@ -90,6 +102,27 @@ public class PublishToPyPiRepoMojo extends AbstractHabushuMojo {
...
+     * retry or set to any negative number for unlimited retries.  Unlimited retries will follow a fibonacci backoff
+     * interval.
+     */
+    @Parameter(property = "habushu.pyPiPushRetries", defaultValue = "3")
+    protected int pypiPushRetries = 3;
+
+    /**
+     * A multiplier to apply to the retry wait time (e.g., round of fibonacci).
+     */
+    @Parameter(property = "habushu.pypiPushRetryMultiplier", defaultValue = "15000")
```
Comment (epwilkins) [inline habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PublishToPyPiRepoMojo.java:117]: A: This one is way more obvious based on the default, but again might be worth saying the unit is `ms`

**Review by epwilkins — APPROVED** [review]

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PublishToPyPiRepoMojo.java (hunk 2 above)
Comment (d-ryan-ashcraft) [reply habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PublishToPyPiRepoMojo.java:123]: Good catch - ✅ FIXED.

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PublishToPyPiRepoMojo.java (hunk 3 above)
Comment (d-ryan-ashcraft) [reply habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PublishToPyPiRepoMojo.java:117]: Good catch - ✅ FIXED.

**Pull request merged**